### DISCUSSION TITLE
How do you customize the 500 error page?

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#28108] [Help]
If something bad happens I don't really want people to see where this is hosted by seeing this error screen, how do I turn it off or make it custom?

https://vercel.com/docs/error/application/FUNCTION_INVOCATION_FAILED

<img width="830" alt="Screen Shot 2021-08-14 at 1 15 21 AM" src="https://user-images.githubusercontent.com/28718/129439931-3cc0ddf4-708e-46b1-9cda-03bbf4eabf81.png">

**Comment by Entkenntnis** [comment]
tried to use custom `_error` but to no avail, see https://github.com/vercel/next.js/discussions/32043

**Reply by icyJoseph** [reply]
But I guess you could create `pages/500.js` right? or `.tsx` 

I wonder if the problem here is with `api` routes, and perhaps those are not caught by the `500.tsx` page, but I'd still give it a try if you guys have a project where the error is happening.

Edit:

It does look like unhandled errors on lambdas will crash out, and show a message like that. Serverless functions seem to not use the custom 500 page. 

I'll assume that the issue is that one of your imports is wrong, perhaps WASM or something like that? Because, otherwise, if feels like there's errors in the lambda execution that are not being caught. 

One thing I've done as a test is:

```js
 // pages/api/hello.js

export default function handler(req, res) {
  try {
    throw new Error('Boom');
    res.status(200).json({ name: 'John Doe' })
  } catch(e){
    res.redirect('/500');
  }
}
```

This is purely for show, I think you should just return `res.status(500).json({message: "Internal Server Error"});` or, simply send the `Internal Server Error` message as text response.

**Reply by Entkenntnis** [reply]
I can create a `pages/500.js`, but this is static. With `_error.js`, I can access the error message and show it on the 500 page.

But yes, the static 500 page is working and we are going into this direction
